### ISSUE TITLE
Rendering issues on demo

### ISSUE THREAD

**Issue by jamesmcm** [tui-rs-revival/ratatui#203]
## Description

The demo does not render correctly (after 1 frame it starts spamming the terminal with the layout code).

See here: 
![2023-05-30-180448_3840x1570_scrot](https://github.com/tui-rs-revival/ratatui/assets/54889/dcac1480-95d3-4f7f-87fb-c70fb51ee949)

I'm using alacritty 0.12.1 (5fdfd47f), and running fish 3.6.1 inside tmux 3.3a on Arch Linux.

The demo fails like this with all backends, but some examples work e.g. list (albeit with some theming issues) : 

![2023-05-30-213245_1275x1573_scrot](https://github.com/tui-rs-revival/ratatui/assets/54889/ba9f7708-16f3-4a18-b00c-ea9cba6845d7)

Config details:
`~/.config/alacritty/alacritty.yml`:
```yaml
window:
  decorations: none
scrolling:
  history: 0
font:
  normal:
    family: "Noto Sans Mono"
  size: 13.0
colors:
  primary:
    background: '0x151515'
    foreground: '0xd0d0d0'
  cursor:
    text: '0x151515'
    cursor: '0xd0d0d0'
  normal:
    black:   '0x151515'
    red:     '0xac4142'
    green:   '0x90a959'
    yellow:  '0xf4bf75'
    blue:    '0x6a9fb5'
    magenta: '0xaa759f'
    cyan:    '0x75b5aa'
    white:   '0xd0d0d0'
  bright:
    black:   '0x505050'
    red:     '0xd28445'
    green:   '0x202020'
    yellow:  '0x303030'
    blue:    '0xb0b0b0'
    magenta: '0xe0e0e0'
    cyan:    '0x8f5536'
    white:   '0xf5f5f5'
draw_bold_text_with_bright_colors: false
hints:
  enabled:
   - regex: "(ipfs:|ipns:|magnet:|mailto:|gemini:|gopher:|https:|http:
   ..|news:|file:|git:|ssh:|ftp:)\
             [^\u0000-\u001F\u007F-\u009F<>\"\\s{-}\\^⟨⟩`]+"
     command: chromium
     post_processing: true
     mouse:
       enabled: true
       mods: None
     binding:
       key: U
       mods: Control|Shift
key_bindings:
  - { key: V,        mods: Alt, action: Paste                            }
  - { key: C,        mods: Alt, action: Copy                             }
  - { key: V,        mods: Control|Shift,    action: ReceiveChar               }
  - { key: C,        mods: Control|Shift,    action: ReceiveChar                }
```

`~/.tmux.conf` :
```conf
# set scroll history to 100,000 lines
set-option -g history-limit 100000

# modern colors
set -g default-terminal "tmux-256color"
set -ga terminal-overrides ",alacritty:Tc"

# unbind the prefix and bind it to Ctrl-a like screen
# unbind C-b
# set -g prefix C-a
# bind C-a send-prefix
bind '"' split-window -c "#{pane_current_path}"
bind % split-window -h -c "#{pane_current_path}"
bind c new-window -c "#{pane_current_path}"

# copy to X clipboard
bind -T copy-mode-vi v send -X begin-selection
bind -T copy-mode-vi y send-keys -X copy-pipe 'xclip -in -selection clipboard &> /dev/null'
bind -T copy-mode-vi Enter send-keys -X cancel

# shortcut for moving tmux buffer to clipboard
# useful if you've selected with the mouse
bind-key -nr C-y run "tmux show-buffer | xclip -in -selection clipboard &> /dev/null"

# Avoid ESC delay
set -s escape-time 0

# Fix titlebar
set -g set-titles on
set -g set-titles-string "#T"

# VIM mode
set -g mode-keys vi

# Mouse friendly
set -g mouse on

# Move between panes with vi keys
bind h select-pane -L
bind j select-pane -D
bind k select-pane -U
bind l select-pane -R

set -g status-right ''
set -g status-right-length 0

source-file ~/.config/base16/tmux/base16-classic-dark.conf
```

`~/.config/base16/tmux/base16-classic-dark.conf`: 
```conf
set-option -g status-style "fg=#B0B0B0,bg=#202020"
set-window-option -g window-status-style "fg=#B0B0B0,bg=default"
set-window-option -g window-status-current-style "fg=#F4BF75,bg=default"
set-option -g pane-border-style "fg=#202020"
set-option -g pane-active-border-style "fg=#303030"
set-option -g message-style "fg=#D0D0D0,bg=#202020"
set-option -g display-panes-active-colour "#90A959"
set-option -g display-panes-colour "#F4BF75"
set-window-option -g clock-mode-colour "#90A959"
set-window-option -g mode-style "fg=#B0B0B0,bg=#303030"
set-window-option -g window-status-bell-style "fg=#202020,bg=#AC4142"
```

## To Reproduce

Above run with:
`cargo run --example demo --release -- --tick-rate 200`
from master with:
`cargo 1.71.0-nightly (ac8401032 2023-05-02)`
`rustc 1.71.0-nightly (74c482104 2023-05-04)`

## Expected behavior

The demo should render correctly.

## Environment

- OS: Linux (Arch Linux)
- Terminal Emulator: Alacritty
- Font: Noto Sans Mono
- Crate version: master
- Backend: crossterm (default)

## Additional context

The strange thing is that it works okay on my laptop with a similar setup (but more recent config), I haven't had time to compare them directly yet though.

**Comment by orhun** [comment]
Are you using the latest revision of `ratatui`? There was a similar bug recently and it was fixed in #198

**Comment by jamesmcm** [comment]
Wow, that was the issue, just bad timing when I cloned the repo before. Thanks!
